pym/bob/generators/VisualStudio.py

```diff
@@ -0,0 +1,320 @@
+        #self.headers.append(str(header.relative_to(self.root)))
+        self.headers.append(str(self.recipesRoot.joinpath(header)))
+    def addSource(self, source):
+        #self.sources.append(str(source.relative_to(self.root)))
+        self.sources.append(str(self.recipesRoot.joinpath(source)))
+    def addResource(self, resource):
+        #self.resources.append(str(resource.relative_to(self.root)))
+        self.resources.append(str(self.recipesRoot.joinpath(resource)))
+    def addIncPath(self, path):
+        self.incPaths.append(str(self.recipesRoot.joinpath(path)))
+    def addDependency(self, dep):
+        self.dependencies.add(dep)
+
+    def generateProject(self, projects, cmd):
+        items = []
+        if self.headers:
+            g = ElementTree.Element("ItemGroup")
+            for i in self.headers: ElementTree.SubElement(g, "ClInclude", {"Include" : str(i)})
+            items.append(ElementTree.tostring(g, encoding="unicode"))
+        if self.sources:
+            g = ElementTree.Element("ItemGroup")
+            for i in self.sources: ElementTree.SubElement(g, "ClCompile", {"Include" : str(i)})
+            items.append(ElementTree.tostring(g, encoding="unicode"))
+        # TODO: self.resources
+        if self.dependencies:
+            g = ElementTree.Element("ItemGroup")
+            incPaths = []
+            for i in sorted(self.dependencies):
+                dep = ElementTree.SubElement(g, "ProjectReference",
+                        {"Include" : "..\\{NAME}\\{NAME}.vcxproj".format(NAME=i)})
+                ElementTree.SubElement(dep, "Project").text = "{" + str(projects[i].uuid) + "}"
+                ElementTree.SubElement(dep, "LinkLibraryDependencies").text = "true"
+                incPaths.extend(projects[i].incPaths)
+
+            items.append(ElementTree.tostring(g, encoding="unicode"))
+
+            n = ElementTree.Element("PropertyGroup", {"Condition" : "'$(Configuration)|$(Platform)'=='Build|Win32'"})
+            ElementTree.SubElement(n, "NMakeOutput")
+            ElementTree.SubElement(n, "NMakeIncludeSearchPath").text = ";".join(incPaths)
+            ElementTree.SubElement(n, "NMakeBuildCommandLine").text = cmd + " -b " + self.packagePath
+            ElementTree.SubElement(n, "NMakeReBuildCommandLine").text = cmd + " -b -f --clean " + self.packagePath
+            # TODO: NMakeCleanCommandLine == "rm -rf dev/{build,dist}"?
+            items.append(ElementTree.tostring(n, encoding="unicode"))
+
+            n = ElementTree.Element("PropertyGroup", {"Condition" : "'$(Configuration)|$(Platform)'=='Checkout+Build|Win32'"})
+            ElementTree.SubElement(n, "NMakeOutput")
+            ElementTree.SubElement(n, "NMakeIncludeSearchPath").text = ";".join(incPaths)
+            ElementTree.SubElement(n, "NMakeBuildCommandLine").text = cmd + " " + self.packagePath
+            ElementTree.SubElement(n, "NMakeReBuildCommandLine").text = cmd + " -f --clean " + self.packagePath
```

Comment:
Here I added -v, I guess in QtCreator plugin -v is also default?
But this is something I can do: bring all build-modes from Qt-Creator plugin to here.

Reply:
fixed.